metrics/perplexity/perplexity.py

```diff
@@ -100,22 +105,28 @@ def _compute(
         else:
             device = "cuda" if torch.cuda.is_available() else "cpu"
 
-        print("cuda available?", torch.cuda.is_available())
 
         model = AutoModelForCausalLM.from_pretrained(model_id)
         model = model.to(device)
 
-        tokenizer = AutoTokenizer.from_pretrained(model_id, pad_token="<PAD>")
+        tokenizer = AutoTokenizer.from_pretrained(model_id)
```

Comment:
To update you all, @TristanThrush, @lhoestq, @sashavor:
The tokenization/special tokens seems to have been the main culprit (@TristanThrush you were right!)

What I've done as a (temporary and probably not optimal) fix for now is to use one of the other existing special tokens so as to not have to expand the vocabulary. In theory, this shouldn't mess up any of the computations (as the attention mask masks out all special tokens anyways, I believe), and indeed gets the outputs to pretty much match (at least to a few decimal places) what I was getting before.

I've also removed stride as per @TristanThrush 's comment, as well as changed it to return both the mean_perplexity and the list of perplexities, as per @lhoestq 's suggestion.